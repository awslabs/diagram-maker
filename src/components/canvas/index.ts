export { default as Canvas, CanvasProps } from './Canvas';
