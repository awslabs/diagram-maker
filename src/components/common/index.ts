export { default as ComposeView } from './ComposeView';
