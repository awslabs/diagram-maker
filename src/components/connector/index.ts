export {
  default as Connector, ConnectorProps, ConnectorType
} from './Connector';
