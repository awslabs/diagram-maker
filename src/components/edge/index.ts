export { EdgeStyle } from './EdgeCurve';
export { default as Edge, EdgeProps } from './Edge';
export { default as EdgeBadge, EdgeBadgeProps } from './EdgeBadge';
export { default as PotentialEdge, PotentialEdgeProps } from './PotentialEdge';
