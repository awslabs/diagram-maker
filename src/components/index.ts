export { Canvas, CanvasProps } from './Canvas';
export { default as ComposeView } from './common/ComposeView';
export {
  default as Connector,
  ConnectorProps,
  ConnectorType
} from './connector/Connector';
export {
  default as ContextMenu,
  ContextMenuProps
} from './contextMenu/ContextMenu';
export { Node, NodeProps, PotentialNode, PotentialNodeProps } from './node';
export { Panel, PanelProps } from './panel';
export { SelectionMarquee, SelectionMarqueeProps } from './selectionMarquee';
export { View, ViewProps } from './view/View';
export { getConnectedView, ConnectedViewOwnProps } from './view/ConnectedView';
export { Workspace, WorkspaceProps } from './Workspace';

export * from './edge/EdgeCurve';
export { default as Edge, EdgeProps } from './edge/Edge';
export { default as EdgeBadge, EdgeBadgeProps } from './edge/EdgeBadge';
export {
  default as PotentialEdge,
  PotentialEdgeProps
} from './edge/PotentialEdge';

export {
  default as GridPattern,
  GridPatternProps
} from './canvas/patterns/GridPattern';

export {
  default as DotPattern,
  DotPatternProps
} from './canvas/patterns/DotPattern';

export { render, destroy } from './renderUtils';
