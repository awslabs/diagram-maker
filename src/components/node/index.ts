export { default as Node, NodeProps } from './Node';
export { default as PotentialNode, PotentialNodeProps } from './PotentialNode';
