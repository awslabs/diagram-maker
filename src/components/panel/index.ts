export { default as Panel, PanelProps } from './Panel';
