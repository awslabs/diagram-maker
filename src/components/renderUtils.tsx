import * as Preact from 'preact';
import { Provider } from 'preact-redux';
import { Store } from 'redux';

import { getConnectedView } from 'diagramMaker/components/view/ConnectedView';
import ConfigService from 'diagramMaker/service/ConfigService';
import { DiagramMakerData } from 'diagramMaker/state/types';

export function render<NodeType, EdgeType>(
  store: Store<DiagramMakerData<NodeType, EdgeType>>,
  container: HTMLElement,
  configService: ConfigService<NodeType, EdgeType>,
  root?: Element
) {
  const ConnectedView = getConnectedView<NodeType, EdgeType>();
  return Preact.render(
    <Provider store={store}>
      <ConnectedView configService={configService} />
    </Provider>,
    container, root
  );
}

export function destroy(container: HTMLElement, root: Element) {
  return Preact.render(null, container, root);
}
