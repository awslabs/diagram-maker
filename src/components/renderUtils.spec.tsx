const originalPreact = jest.requireActual('preact');
const mockPreact = {
  ...originalPreact,
  render: jest.fn()
};

jest.mock('preact', () => mockPreact);

import * as Preact from 'preact';

import { destroy, render } from 'diagramMaker/components/renderUtils';

describe('renderUtils', () => {
  const store = {} as any;
  const container = 'someContainer' as any;
  const config = 'someConfig' as any;
  const root = 'someDom' as any;

  beforeEach(() => {
    mockPreact.render.mockClear();
  });

  describe('render', () => {
    it('calls Preact.render passing container & root', () => {
      render(store, container, config, root);
      expect(Preact.render).toHaveBeenCalledWith(expect.anything(), container, root);
    });

    it('renders view', () => {
      render(store, container, config, root);
      expect(mockPreact.render.mock.calls[0][0]).toMatchSnapshot();
    });
  });

  describe('destroy', () => {
    it('calls Preact.render with null passing container & root', () => {
      destroy(container, root);
      expect(Preact.render).toHaveBeenCalledWith(null, container, root);
    });
  });
});
