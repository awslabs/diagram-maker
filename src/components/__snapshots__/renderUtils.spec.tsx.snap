// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderUtils render renders view 1`] = `
<Provider store={Object {}}>
  <Connect(View) configService="someConfig"></Connect(View)>
</Provider>
`;
